src/components/Logout/LogoutScreen.test.tsx

```diff
@@ -0,0 +1,298 @@
+  withNavigationFocus: (Component: FunctionComponent) => (props: any) => (
+    <Component navigation={{ navigate: mockNavigate }} {...props} />
+  ),
+}));
+
+const mockCaptureException = Sentry.captureException as jest.MockedFunction<
+  typeof Sentry.captureException
+>;
+const mockNavigate = jest.fn() as jest.MockedFunction<
+  NavigationScreenProp<unknown>["navigate"]
+>;
+const mockCallLogout = callLogout as jest.MockedFunction<typeof callLogout>;
+
+const LogoutScreenContainer = withNavigation(LogoutScreen);
+
+const keyA = "keyA";
+const keyB = "keyB";
+const keyC = "keyC";
+
+const authA: AuthCredentials = {
+  endpoint: "endA",
+  expiry: 0,
+  operatorToken: "optA",
+  sessionToken: "sesA",
+};
+const authB: AuthCredentials = {
+  endpoint: "endB",
+  expiry: 0,
+  operatorToken: "optB",
+  sessionToken: "sesB",
+};
+const authC: AuthCredentials = {
+  endpoint: "endC",
+  expiry: 0,
+  operatorToken: "optC",
+  sessionToken: "sesC",
+};
+
+const authCredentials = {
+  [keyA]: authA,
+  [keyB]: authB,
+  [keyC]: authC,
+};
+
+const defaultAuthStoreContextValue = {
+  authCredentials: authCredentials,
+  clearAuthCredentials: jest.fn(),
+  hasLoadedFromStore: true,
+  removeAuthCredentials: jest.fn(),
+  setAuthCredentials: jest.fn(),
+};
+
+const defaultCampaignConfigStoreContextValue = {
+  allCampaignConfigs: {},
+  clearCampaignConfigs: jest.fn(),
+  hasLoadedFromStore: true,
+  removeCampaignConfig: jest.fn(),
+  setCampaignConfig: jest.fn(),
+};
+
+const defaultConfigContextProviderValue = {
+  config: { appMode: AppMode.production },
+  setConfigValue: jest.fn(),
+};
+
+describe("LogoutScreen", () => {
+  beforeEach(() => {
+    mockCaptureException.mockReset().mockName("captureException");
+    mockNavigate.mockReset().mockName("navigate");
+    mockCallLogout.mockReset().mockName("callLogout");
+  });
+
+  it("should attempt logout from all campaigns", async () => {
+    expect.assertions(5);
+
+    render(
+      <ConfigContext.Provider value={defaultConfigContextProviderValue}>
+        <AuthStoreContext.Provider value={defaultAuthStoreContextValue}>
+          <CampaignConfigsStoreContext.Provider
+            value={defaultCampaignConfigStoreContextValue}
+          >
+            <ImportantMessageSetterContext.Provider value={jest.fn()}>
+              <LogoutScreenContainer />
+            </ImportantMessageSetterContext.Provider>
+          </CampaignConfigsStoreContext.Provider>
+        </AuthStoreContext.Provider>
+      </ConfigContext.Provider>
+    );
+
+    await waitFor(() => expect(mockNavigate).toHaveBeenCalledTimes(1));
+
+    expect(mockCallLogout).toHaveBeenCalledTimes(3);
+    expect(mockCallLogout).toHaveBeenCalledWith("sesA", "optA", "endA");
+    expect(mockCallLogout).toHaveBeenCalledWith("sesB", "optB", "endB");
+    expect(mockCallLogout).toHaveBeenCalledWith("sesC", "optC", "endC");
+  });
+
+  it("should logout from campaigns with no error and vice versa", async () => {
```

Comment:
This description can probably be improved? It’s very ambiguous now. Maybe 

```
should logout from all other campaigns even when one of them throws
```

Or better,

```
describe(“when logging out of one of the campaigns throws error”, () => {
  it(“should log out of all other campaigns”, () => {…});
});
```

but whether to do the first or second, I’ll leave it up to you. Better to have consistency within the file too.